pkg/common/k8s.go

```diff
@@ -262,3 +266,133 @@ func EnrichProbe(probe **v1.Probe, defaultProbe *v1.Probe) {
+	}
+
+	return serviceAccount, nil
+}
+
+func GetProjectSecret(ctx context.Context, kubeClient kubernetes.Interface, projectSecretTemplate, projectName, namespace string) (*v1.Secret, error) {
+	// if project secret template is not specified, return empty data
+	if projectSecretTemplate == "" {
+		return nil, nil
+	}
+
+	// render the project secret name using the template
+	templateData := map[string]interface{}{
+		"ProjectName": projectName,
+		"Namespace":   namespace,
+	}
+	secretName, err := renderProjectSecretName(projectSecretTemplate, templateData)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to render project secret name")
+	}
+
+	// fetch the secret from Kubernetes
+	projectSecret, err := kubeClient.CoreV1().Secrets(namespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		// if not found, skip validation
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, errors.Wrapf(err, "Failed to get secret %s", secretName)
+	}
+	return projectSecret, nil
+}
+
+func IsServiceAccountAllowed(secret *v1.Secret, secretAllowedServiceAccountsKey string, serviceAccount string) error {
+
+	allowedServiceAccounts, found := getAllowedServiceAccountsFromSecret(secret, secretAllowedServiceAccountsKey)
+
+	// if the key is found, but is empty, treat it as no allowed service accounts configured
+	if len(allowedServiceAccounts) == 0 && found {
+		return errors.Errorf("No service accounts are allowed")
+	}
+	if len(allowedServiceAccounts) == 0 {
+		return nil
+	}
+	// trim spaces and check membership
+	requestedSA := strings.ToLower(strings.TrimSpace(serviceAccount))
+	for _, sa := range allowedServiceAccounts {
+		if strings.ToLower(strings.TrimSpace(sa)) == requestedSA {
+			return nil
+		}
+	}
+
+	return errors.Errorf("Service account %q is not allowed", requestedSA)
+}
+
+func EnrichServiceAccount(secret *v1.Secret, secretDefaultServiceAccountsKey, serviceAccount, defaultPlatformServiceAccount string) string {
+	if serviceAccount != "" {
+		return serviceAccount
+	}
+	defaultProjectSa, _ := GetStringValueFromSecret(secret, secretDefaultServiceAccountsKey)
+	if defaultProjectSa == "" {
+		return defaultPlatformServiceAccount
+	}
+	return defaultProjectSa
+}
+
+// getAllowedServiceAccountsFromSecret retrieves the allowed service accounts from the secret
+// It returns a slice of allowed service accounts and a boolean indicating if the key was found in the secret
+func getAllowedServiceAccountsFromSecret(secret *v1.Secret, secretAllowedServiceAccountsKey string) (allowedServiceAccounts []string, found bool) {
+	if secret == nil || secretAllowedServiceAccountsKey == "" {
+		return
+	}
+
+	allowed, ok := GetStringValueFromSecret(secret, secretAllowedServiceAccountsKey)
+	if !ok {
+		// if the key is not found, return empty slice
+		return
+	}
+
+	// if the key is found, set found to true and split the string by comma
+	found = true
+	allowedServiceAccounts = strings.Split(allowed, ",")
```

Comment:
Why not do the `TrimSpace` and `ToLower` here as well, rather than depend on the caller taking care of that?